src/exit/exit.c

```diff
@@ -12,13 +12,17 @@ weak_alias(dummy, __funcs_on_exit);
 weak_alias(dummy, __stdio_exit);
 weak_alias(dummy, _fini);
 
+#ifndef __CHEERP__
 extern weak hidden void (*const __fini_array_start)(void), (*const __fini_array_end)(void);
+#endif
```

Comment:
What's the rationale, these are not called, what's the problem in leaving the code and avoid conflicts on future rebases?

Reply:
Right, declaration should stay.
Usage in libc_exit_fini though might be problematic, since if the function is not eliminated (and in practice it might survive) then Cheerp will fail in LinearMemoryHelper since it's not robust to non-defined globals.

One possible solution is add handling (eg. compiling constant or equivalent) for undefied globals, or since my impression is that fini_array_* are used only for dynamic linking they are not relevant for Cheerp

Reply:
How can it not be eliminated? How is calling that?